float64[] goal
int32 avail
---
string run
